plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/XmlSourceValidatorTest.java

```diff
@@ -56,18 +56,24 @@
 
   private final IncrementalReporter reporter = new IncrementalReporter(null);
 
-  @Rule public TestProjectCreator nonFacetedProject = new TestProjectCreator();
+  @Rule public TestProjectCreator projectCreator = new TestProjectCreator();
```

Comment:
Do we actually need real projects?  Can we use `IProject` mocks?

Reply:
I'm afraid not. One main reason is to create an `IFile`, which I believe we haven't found a way to mock it in these tests that actually reads contents.

Another complications is that we make use of a lot of actual components which accepts `IProject` and `IFile`, and they interact. They would work properly only when they are real projects and files:

```java
    IWorkbench workbench = PlatformUI.getWorkbench();
    WorkbenchUtil.openInEditor(workbench, file);
    ITextViewer viewer = ValidationTestUtils.getViewer(file);
```
```java
    IncrementalHelper helper = new IncrementalHelper(document, project);
```
```java
      IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
      IFile file = root.getFile(path);
```